index.js

```diff
@@ -1142,7 +1142,17 @@ Tree.prototype.removeNode = function (obj) {
  */
 Tree.prototype.filter = function (fn) {
   if (fn == null) {
-    return this.select((this._selected && this._selected.id) || (this.options.forest ? this.root[0].id : this.root.id), {
+    let selected = this._selected && this._selected.id
+    if (typeof selected !== 'number') {
```

Comment:
Is there any chance the selected ID could be a String?  Maybe if a node is added with a String ID?

Reply:
I don't know, but I'll change it just in case.